AirCastingTests/SearchAndFollowTests/SessionForLocationDownloaderTests.swift

```diff
@@ -0,0 +1,87 @@
+// Created by Lunar on 02/08/2022.
+//
+import XCTest
+import Resolver
+@testable import AirCasting
+
+class SessionForLocationDownloaderTests: ACTestCase {
+    lazy var sut = SessionsForLocationDownloaderDefault()
+    let clientSpy = APIClientSpy()
+    
+    func test_whenRequestingDataFromServer_shouldUseGetMethod() {
+        Resolver.test.register { self.clientSpy as APIClient }
+        MockedSession.getAnySession(using: sut)
+        XCTAssertEqual(clientSpy.request.httpMethod, "GET")
+    }
+    
+    func test_whenCreatingQueryItems_shouldUseSmallQAsKey() {
+        Resolver.test.register { self.clientSpy as APIClient }
+        MockedSession.getAnySession(using: sut)
+        guard let queryItems = URLComponents(string: clientSpy.request.description)?.queryItems else { XCTFail("Problem when trying to get query items from url."); return }
+        XCTAssertEqual(queryItems[0].name, "q")
```

Comment:
no guarantee that there will be a `[0]` element and when not, it will crash. Use an optional `first` with combination with `XCTUnwrap`